glide-core/src/client/reconnecting_connection.rs

```diff
@@ -268,4 +307,14 @@ impl ReconnectingConnection {
             ConnectionState::Reconnecting
         )
     }
+
+    pub async fn wait_for_disconnect_with_timeout(&self, max_wait: &Duration) {
+        // disconnect_notifier always exists
+        self.connection_options
```

Comment:
If it can't be `None` -> lets not use `Option<..>`

Reply:
We cant - its part of the GlideConnectionOptions